### PR TITLE
CLE-NST-AI tests for iterables

### PR DESCRIPTION
**Pull request by mikhailZakharevich**
Here's my current understanding of the task.

### CODE REVIEW

File: src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import org.scalacheck.Prop._
+import io.circe.syntax._
+import io.circe.parser
+import io.circe.{Json, JsonObject, Codec}
+import group.scala.karazin.circe.literal.extras.arbitraries.instances.{given, _}
+
+/** !!! by default, scalacheck generates empty container in `forAll` !!! */
+class IterableEncodeSuite extends munit.ScalaCheckSuite:
+
+  override def scalaCheckTestParameters =
```
Comment (VLADISLAV008) [inline src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala]: Why is this parameter overridden? After all, we want to test for different values.

File: src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    super.scalaCheckTestParameters
+      .withMinSuccessfulTests(1)
+
+  def genIterableOf[T: Arbitrary] =
```
Comment (VLADISLAV008) [inline src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala]: I think this generator is not needed.

File: src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala (hunk 1 above)
Comment (mikhailZakharevich) [reply src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala]: Fixed, although tests take much more time to complete now.

File: src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala (hunk 2 above)
Comment (mikhailZakharevich) [reply src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala]: Done

File: src/test/scala/group/scala/karazin/circe/literal/extras/arbitraries/JsonArbitraries.scala [hunk 3]
```diff
@@ -13,4 +13,7 @@ trait JsonArbitraries {
 
   given Arbitrary[JsonObject] = arbitraryJsonObject
 
+  given Arbitrary[Iterable[JsonObject]] =
```
Comment (VLADISLAV008) [inline src/test/scala/group/scala/karazin/circe/literal/extras/arbitraries/JsonArbitraries.scala]: Is this arbitrary necessary?

Comment (mikhailZakharevich) [reply src/test/scala/group/scala/karazin/circe/literal/extras/arbitraries/JsonArbitraries.scala]: Good point, forgot to remove.
Done.

File: src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      inline def encode(inline args: Any*): Json =
+        ${macros.encode[Iterable[Option[Primitive]]]('sc, 'args)}
+
+    forAll { (optionIterable: Iterable[Option[Primitive]]) =>
+
+      val jsonObjectsArray: Iterable[Json] = optionIterable.map(_.asJson)
+
+      val result: Json = encode"$optionIterable"
+
+      val expected: Json = Json.fromValues(jsonObjectsArray)
```
Comment (VLADISLAV008) [inline src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala]: Maybe it is better to implement it like this
`val expected: Json = optionIterable.asJson`

Comment (mikhailZakharevich) [reply src/test/scala/group/scala/karazin/circe/literal/extras/IterableEncodeSuite.scala]: Yep, that works.

**Review by VLADISLAV008 — APPROVED** [review]

**Review by IgorWolkov — APPROVED** [review]

**Pull request merged**